src/main/java/com/microsoft/sqlserver/jdbc/SQLServerClob.java

```diff
@@ -381,9 +397,8 @@ public String getSubString(long pos, int length) throws SQLException {
      */
     public long length() throws SQLException {
         checkClosed();
-
-        if (value == null && activeStreams.get(0) instanceof PLPInputStream) {
-            return (long) ((PLPInputStream) activeStreams.get(0)).payloadLength / 2;
+        if (null == value && activeStreams.get(0) instanceof BaseInputStream) {
+            return (long) ((BaseInputStream) activeStreams.get(0)).payloadLength;
         }
         return value.length();
```

Comment:
check for null? value could be null here

Reply:
If value is null, it means the lob object is being streamed and the if condition above will return the length of the stream. Else, it means the lob object is loaded into memory and value won't be null.